src/rogue/interfaces/memory/Block.cpp

```diff
@@ -282,8 +282,8 @@ void rim::Block::startTransaction(uint32_t type, bool forceWr, bool check, rim::
          count = retryCount_;
 
       } catch ( rogue::GeneralError err ) {
-         if ( (count+1) >= retryCount_ ) throw err;
-         bLog_->error("Error on try %" PRIu32 " out of %" PRIu32 ": %s", (count+1), (retryCount_+1), err.what());
+         if ( (count+1) > retryCount_ ) throw err;
+         bLog_->error("Error on try %" PRIu32 " out of %" PRIu32 ": %s", (count+1), (retryCount_), err.what());
          fWr = true; // Stale state is now lost
```

Comment:
Should this maybe be
```C++
bLog_->error("Error on try %" PRIu32 " out of %" PRIu32 ": %s", (count+1), (retryCount_), err.what());
fWr = true; // Stale state is now lost
if ( (count+1) > retryCount_ ) throw err;
```

I think we want to log the error and set the stale state regardless, and then only re-throw the exception if this is the final try. 

@slacrherbst what do you think?